Sources/DualCameraKit/DualCameraController.swift

```diff
@@ -0,0 +1,289 @@
+    }
+    
+    /// Cancels all active stream tasks.
+    private func cancelRendererTasks() {
+        for task in streamTasks.values {
+            task.cancel()
+        }
+        streamTasks.removeAll()
+    }
+    
+    public func capturePhotoWithLayout(_ layout: CameraLayout, containerSize: CGSize) async throws -> UIImage {
+        // Get the current screen size (instead of hardcoding)
+        guard let screenSize = await UIApplication.shared.windows.first?.bounds.size else {
+            throw DualCameraError.captureFailure(.unknownDimensions)
+        }
+        
+        // Get frames directly from renderers (which already handle aspect correction)
+        guard let frontRenderer = renderers[.front],
+              let backRenderer = renderers[.back] else {
+            throw DualCameraError.captureFailure(.noPrimaryRenderer)
+        }
+        
+        let frontImage = try await frontRenderer.captureCurrentFrame()
+        let backImage = try await backRenderer.captureCurrentFrame()
+        
+        return renderImagesWithLayout(front: frontImage, back: backImage, layout: layout, screenSize: screenSize)
+    }
+    
+    /// Gets the latest pixel buffer from a camera stream
+    private func getLatestPixelBuffer(for source: CameraSource) async throws -> CVPixelBuffer {
+        let stream = source == .front ? frontCameraStream : backCameraStream
+        
+        // Create a async/await wrapper for getting the next buffer
+        return try await withCheckedThrowingContinuation { continuation in
+            Task {
+                var latestBuffer: CVPixelBuffer?
+                
+                // Wait for the next frame (or use a small timeout)
+                for await wrapper in stream {
+                    latestBuffer = wrapper.buffer
+                    break
+                }
+                
+                if let buffer = latestBuffer {
+                    continuation.resume(returning: buffer)
+                } else {
+                    continuation.resume(throwing: DualCameraError.captureFailure(.noFrameAvailable))
+                }
+            }
+        }
+    }
+    
+    /// Converts a CVPixelBuffer to UIImage
+    private func pixelBufferToUIImage(_ pixelBuffer: CVPixelBuffer) -> UIImage {
+        let ciImage = CIImage(cvPixelBuffer: pixelBuffer)
+        let context = CIContext()
+        let cgImage = context.createCGImage(ciImage, from: ciImage.extent)!
+        return UIImage(cgImage: cgImage)
+    }
+    
+    private func renderImagesWithLayout(front: UIImage, back: UIImage, layout: CameraLayout, screenSize: CGSize) -> UIImage {
+        let renderer = UIGraphicsImageRenderer(size: screenSize)
+        
+        return renderer.image { ctx in
+            // Background fill
+            UIColor.black.setFill()
+            ctx.fill(CGRect(origin: .zero, size: screenSize))
+            
+            switch layout {
+            case .fullScreenWithMini(let miniCamera, let position):
```

Comment:
have only tested this `fullScreenWithMini` and `.bottomTrailing` layout. either test fully here or create f/u issue